### ISSUE TITLE
Improve output for ILL system & link up to Imperial relay script

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#544]
As discussed over email.

**Comment by markmacgillivray** [comment]
These two are irrelevant being sent by us because if always the same they can be decided by the script at the Imperial side (we can set them to be whatever value they tell us to set them to, but that is a pretty circuitous way round) :
* The system will assume they'd like an article
* We'll request digital delivery

This is added:
We'll pass the URL the user was on into a field of your choice

These three are added whenever found:
* DOI will be filled with DOI
* Title will be found by DOI look ups
* Journal title will be found by DOI look up

Not sure what this one means beyond what is above (we don't return the full crossref data because we don't return full results of all our sources in any of our apps yet, see below. Could add the full crossref data, but unless there is clear need it is a bit of extra work):
* Publication data will be found by DOI look up

The Primo search results that our system finds are now also returned, as formatted by our system, although not sure what extra use they will be, and they may not be all results because of paging and so on - there is no need to lookup multiple pages so we lookup the minimum to find a match. Looking up more could be slower and potentially very large, again for unclear benefit as to what it could be used for. As to the search for OA copies, we don't actually collect and return all that in any of our tools, we just return when we do find something and the key metadata (title, doi). To return all possible things we did search whether we found anything or not would be quite a big change to current process, so unless there is clear benefit to knowing we didn't find things everywhere we looked, it is best avoided:
* We'll pass the result of a search of the Imperial Catalog & a search for OA copies into a field of your choice

Where we pass the primo search data, it is in an object called primo that can have keys "title" and "journal" which each point to a result set. But if either of them returned nothing, they won't be present.

Work to solve the journal subscription matching continues. 

Assigning to @JosephMcArthur for info for the followup email to Imperial
